docs/kbs_protocol.md

```diff
@@ -0,0 +1,276 @@
+# KBS Protocol
+
+As a trusted service, KBS's client (Key Broker Client, KBC for short) runs in a HW-TEE. 
+When the KBC sends a request to KBS, HW-TEE attestation is required. Universal-KBS performs a 
+simple, universal and extensible "Request-Challenge-Attestation-Response" (we call it **RCAR** 
+for short) protocol to complete the attestation, and encrypt the response payload content 
+returned by KBS to KBC with the key which protected by HW-TEE.
+
+# Introduction
+
+The purpose of the attestation between KBS and KBC is actually to confirm whether 
+**the platform where the KBC is located is in the expected security state**, that is, it runs in 
+a real and harmless HW-TEE. Therefore, naturally, HW-TEE authentication can be regarded as an 
+extension/enhancement of the existing security authentication protocol stack. At present, some 
+projects have made such attempts, such as [HTTPA](https://arxiv.org/abs/2110.07954), [RATS-TLS](https://github.com/inclavare-containers/rats-tls) and so on. 
+However, Universal-KBS do not intend to apply these projects directly, nor will we design a new 
+similar implementation, because it is inevitable to make some deep-seated adaptation to the 
+existing transmission module at the server and client. We hope that the intrusion of 
+Universal-KBS into existing transmission modes will be as few as possible, so that Universal-KBS 
+can be applied in more and broader scenarios.
```

Comment:
I'm not sure this is necessary. I guess it might be good to acknowledge other methods, but I think the purpose of this document is to define our protocol rather than compare it with other things or try to justify it.